### ISSUE TITLE
Kore try to change party setting every about 30 seconds

### ISSUE THREAD

**Issue by phamtrong204** [OpenKore/openkore#870]
###### Summary: 
My Kore try to change party setting every about 30 seconds!
But there is no need to change!

###### Affected configuration(s)/ file(s):
```
partyAuto 2
partyAutoShare 1
partyAutoShareItem 0
partyAutoShareItemDiv 0
```
###### Actual Behavior:
<img width="334" alt="screen shot 2017-08-08 at 09 34 30" src="https://user-images.githubusercontent.com/6489256/29054978-d34e7422-7c23-11e7-98e9-fb3b8add99fa.png">

**Comment by alisonrag** [comment]
party share are sended to server everytime when we receive the following packets:
'00FB' => ['party_users_info']
'01E9' => ['party_join']

we should verify if is not set before send

**Comment by moneyisgod** [comment]
How about checking if we are the party leader?

**Comment by alisonrag** [comment]
kore check everytime for this params:

`	if (($config{partyAutoShare} || $config{partyAutoShareItem} || $config{partyAutoShareItemDiv}) && $char->{party} && %{$char->{party}} && $char->{party}{users}{$accountID}{admin}) {
		$messageSender->sendPartyOption($config{partyAutoShare}, $config{partyAutoShareItem}, $config{partyAutoShareItemDiv});
	}`

**Comment by alisonrag** [comment]
#874 is working fine
Maybe we can add a timeout to ensure that Kore will not send the party share command every time when he are not able (for example when difference of level of party member are to high)

**Comment by phamtrong204** [comment]
I think it already has a timeout.
It already does not spam send share command to the server when it not able.
And your commit adds more condition, which can only reduce the frequency of the request.

**Comment by c4c1n6kr3m1** [comment]
The code is on the wrong place

Should be inside "party_exp"

**Comment by alisonrag** [comment]
@phamtrong64
Kore dont have timeout, he send everytime when receive the following packets
00FB' => ['party_users_info']
'01E9' => ['party_join']

@c4c1
party_exp is received ONLY if kore send party Share setting
If you want to move this function to inside of party_exp i see 2 possibility:
1 - never share exp
2 - possible loop/spam

**Comment by c4c1n6kr3m1** [comment]
@alisonrag
you always receive party_exp every time you logon

rather than double code you made for "party_join" and "party_users_info", you only need 1 if you move it to party_exp and yes, you need to becareful with party level error

**Comment by alisonrag** [comment]
@c4c1n6kr3m1
Sorry, my fault i forgot about the packet when you login, but a didn't made double code, it was there before, i just adjusted the conditions to send the command.

If we move the code to party_exp what we will do when someone join in the party?
Maybe we can move, for now, to function when someone join in the party and to party_exp

**Comment by c4c1n6kr3m1** [comment]
when a leader login and receive party_exp, he will change party setting once ( only if not in "share mode" )

when a party already on "share mode", if someone join ( in acceptable level ) than nothing's gonna happen to party setting

**Comment by alisonrag** [comment]
need to test, but if you are right i will remove the code from both functions and move to party_exp

thanks for your help

**Comment by c4c1n6kr3m1** [comment]
uh, actually , i never face that situation
i share only when level is acceptable, since begining
i believe you need macro or typing manual command  to make your party member move to other map , than get back when the level is on sharing level , and need to make the leader disable sharing first or it will more difficult to catch up the level.

**Comment by alisonrag** [comment]
I tryied to move the party share to party_exp, but if you send a console command to set the exp share, kore will send the command to return to config default.

best way is move to AI and add ai command party share 0,1,2 or 0,1,auto, where 2 or auto are to use AI and config.txt to define the share

**Comment by c4c1n6kr3m1** [comment]
well,forgot to tell you about that
i changed my  command so whenever i type manual share, than it will modify the $config{partyAutoShare} too

**Comment by alisonrag** [comment]
fixed in #886
